### ISSUE TITLE
About calculating parameters of BN

### ISSUE THREAD

**Issue by AITTSMD** [timgaripov/swa#7]
Hi, Thanks for your great work. In your paper, you say after training the model, we should run one additional pass over the train data to get the running mean and standard deviation. I don't know why we should do that. What about the  gamma, beta needed in BN?

**Comment by izmailovpavel** [comment]
Hi, sorry for the late reply. Batchnorm approximates activation statistics throughout training. However, with SWA, we never use the SWA solution during training. SWA weights are calculated separately from training, and the network with SWA weights is never used on data, so no batchnorm statistics are computed for this network. For this reason, we need to somehow approximate the batchnorm statistics separately from training. To do so, we directly compute the mean and variance of activations in one forward pass with the SWA network. 

Also see the description here:
https://pytorch.org/blog/stochastic-weight-averaging-in-pytorch/#batch-normalization
